### DISCUSSION TITLE
Recording accurately frame by frame

### DISCUSSION THREAD

**Discussion by galipmedia** [cables-gl/cables_docs#521] [Answers & Questions]
Hi everyone, I am new to cables.gl, its so cool. My aim is to try to use it to make a kind of video editor/compositor where the user can toggle on and off effects, and have some music mixed with original audio and some gl transitions if possible. 

In my case the recording needs to be precise because on slower pcs it will not be good enough to use a recorder that may skip frames.

So i built this... https://cables.gl/edit/Ey7FHt in an attempt to do what I wanted. It has a modified version of the media recorder but uses a webm muxer (so the final video has a time track). Its very rough but I am just throwing it out there to see if anyone is interested to help get it working correctly. At the moment it still doesn't sync the the sound well for me. It is very hit and miss on my terrible old pc.

I would greatly appreciate anyone more knowledgeable in cables.gl to take a look and see how it could be tightly synced.

**Comment by steam0r** [comment]
do you need this "once" or for every user of the patch? if you only need it once it might make sense to use RenderAnim, download textures for every frame and do the audio-sync in another tool. mediarecorder is just pushing this off to the browser (and hence out of our control).

https://cables.gl/op/Ops.Gl.RenderAnim_v2

there is also a video tutorial on mediarecorder with a section regarding audio, so you might want to have a look at that (if you didn't already):

https://www.youtube.com/watch?v=zWfwC-GH7qk

**Comment by galipmedia** [comment]
In my experience testing the mediaRecorder on my slow old pc I couldn't get high frame rate recordings. I am layering 1920 x 1080 videos with webgl effects etc so it's not surprising it can't record so well in realtime. My users are gamers and are very particular about the quality of their videos. That's why I decided to make a frame-by-frame recorder. It works very well for recording video at exact frame rates, it is also correctly muxed so when played back it has correct duration displayed in the player, which is important in my use case. The only issue is as you can see by the attached examples the sound is inconsistent in speed. Something to do with timestamps, its a pity because it wouldn't require any serverside ffmpeg if it worked.

[comparison.zip](https://github.com/cables-gl/cables_docs/files/11459067/comparison.zip)

I have tried the default audioRecorder but it seems very low quality for some reason. The best solution for sound is the mediaRecorder however it fails when you choose audio only option.

![image](https://github.com/cables-gl/cables_docs/assets/1904800/fd65051a-5826-41aa-83de-b7b182767b44)

**Reply by steam0r** [reply]
thanks for the reply. i looked into it and fixed the audio-only mode. you can try on dev.cables.gl and give it another shot.
